src/quicktube.js

```diff
@@ -21,89 +26,113 @@ const trackEvent = (event) => {
     }
 };
 
-const getCurrentSegment = (currentPosition, duration) => {
+const createPlayerUrl = (playerEmbedUrl, playerId, options) => {
+    let url = `${playerEmbedUrl}${playerId}?autoplay=1`;
+    const optionKeys = Object.keys(options);
+    optionKeys.forEach((key) => {
+        url += `&${key}=${options[key]}`;
+    });
+    return url;
+};
+
+const getCurrentSegment = (currentPosition, duration, numberOfSegments = 4) => {
     const percentage = (currentPosition / duration);
+    // Ensure value is rounded to nearest whole segment eg. 1, 2, 3 , 4
     return (Math.floor(percentage * numberOfSegments) / numberOfSegments).toFixed(2);
 };
```

Comment:
Not sure what exactly you want me to split when you say "we're starting to get a few"